doc/spec_InterruptControllerHandler.dox

```diff
@@ -0,0 +1,91 @@
+// put the namespace around the doxygen block so we don't have to give it all the time in the code to get links
+namespace ChimeraTK {
+/**
+\page spec_InterruptHandler Technical specification: InterruptHandler V0.1draft
+
+> **NOTICE FOR FUTURE RELEASES: AVOID CHANGING THE NUMBERING!** The tests refer to the sections, incl. links and unlinked references from tests or other parts of the specification. These break, or even worse become wrong, when they are not changed consistenty!
+
+## Introduction ##
+
+This document describes the concept of interrupt handlers in backends, their implementation and their interface to firmware, i.e. their description in the map file.
+
+### Context ###
+
+The current implementation INTERRUPTc:i (with c being the interrupt controller number and i being the interrupt) is wrong on several levels.
+* The highest level is always associated with a hardware interrupt. If there is one interrupt controller, this is correct. If there is no interrupt controller, it is already wrong. It currently is INTERRUPT0:i, but should be INTERRUPTi. (e.g. with interrupt controller INTERRUPT0:5 is on HW interrupt 0, while without it is on HW interrupt 5, which is inconsistent).
+* Nested interrupts cannot be described at all.
+
+This specification describes the hierarchical interrupt scheme that solves these issues.
+
+## A. Definitions ##
```

Comment:
For me, a section "Definitions" would describe terms used by the spec to describe the behaviour. These definitions seem to describe already the implementation, as the keyword "INTERRUPT" is probably found in the map file. I think, at this point of the specification we need to describe the behaviour not the implementation. So what should be defined here are terms like "interrupt controller" and "interrupt handler" etc. (as they are used here in this document).

Also, I think it makes it complicated by weaving the concept of nested interrupts strongly into the fundamental principles and wording. I would start to introduce the concept of a single interrupt (aka "INTERRUPTi" but ideally without the map file syntax), then based on that introduce the concept of an interrupt controller (aka "[baseInterrupt]:k", again ideally free of the map file syntax). The concept of nested interrupts then follows natively essentially by induction.

My second paragraph here in this comment is actually more directed already towards the description of the behaviour. In this Definitions section we should only concentrate on the definitions which are used below and merely describe what they mean (not how they interact).